pyanaconda/ui/gui/spokes/custom_storage.py

```diff
@@ -1219,61 +1223,59 @@ def on_remove_clicked(self, button):
-                        self._storage_playground.reset_device(device)
+                    self._storage_playground.reset_device(device_name)
                 else:
                     # Destroying a non-existing device doesn't require any
                     # confirmation.
-                    self._destroy_device(device)
+                    self._destroy_device(device_name)
             else:
                 # This is a device that exists on disk and most likely has data
                 # on it.  Thus, we first need to confirm with the user and then
                 # schedule actions to delete the thing.
                 # In multiselection user could confirm once for all next
                 # selections.
                 if not option_checked:
-                    (rc, option_checked) = self._show_confirmation_dialog(
-                        root_name, device, protected_types
-                    )
+                    (rc, option_checked) = self._show_confirmation_dialog(root_name, device_name)
 
                     if rc != 1:
                         if option_checked:
                             break  # skip evaluation of all other mountpoints
                         continue
 
                 if option_checked and not is_multiselection:
-                    otherpgs = (pg for pg in self._accordion.all_pages
-                                if pg is not page)
+                    otherpgs = (pg for pg in self._accordion.all_pages if pg is not page)
                     otherdevs = []
                     for otherpg in otherpgs:
-                        otherdevs.extend(mem._device.id for mem in otherpg.members)
+                        otherdevs.extend(s.device_name for s in otherpg.members)
                     # We never want to delete known-shared devs here.
                     # The same rule applies for selected device. If it's shared do not
                     # remove it in other pages when Delete all option is checked.
-                    for dev in (s._device for s in page.members
-                                if s._device.id not in otherdevs):
+                    for device in (s.device_name for s in page.members
+                                   if s.device_name not in otherdevs):
                         # we only want to delete boot partitions if they're not
                         # shared *and* we have no unknown partitions
-                        if not self._get_unused_devices() or dev.format.type not in protected_types:
-                            self._destroy_device(dev)
+                        if not self._get_unused_devices() or device.format.type not in \
+                                protected_types:
+                            self._destroy_device(device)
                         else:
-                            log.debug("Device %s cannot be removed.", dev.name)
+                            log.debug("Device %s cannot be removed.", d)
```

Comment:
d should be device?
and perhaps device should be device_name in the for loop?

Reply:
Fixed.